packages/widget.client-react/src/utils/transformEmitEventPayload.test.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2025 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Client, ObjectSet } from "@osdk/client";
+import { createAndFetchTempObjectSetRid } from "@osdk/client/internal";
+import type { ObjectType } from "@osdk/widget.api";
+import { defineConfig } from "@osdk/widget.client";
+import type { Mock } from "vitest";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { transformEmitEventPayload } from "./transformEmitEventPayload.js";
+
+vi.mock("@osdk/client/internal", () => ({
+  createAndFetchTempObjectSetRid: vi.fn(),
+}));
+
+describe("transformEmitEventPayload", () => {
+  const client = vi.fn() as Mock<Client> & Client;
+
+  const createMockObjectType = (rid = "ri.object-type.123"): ObjectType => ({
+    apiName: "MyObjectType",
+    type: "object",
+    internalDoNotUseMetadata: { rid },
+  });
+
+  const mockObjectSet = {} as ObjectSet;
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("should transform ObjectSet parameters and pass through others", async () => {
```

Comment:
Should there be another case for full pass through without any object set parameters?